src/gc/impl/conservative/gc.d

```diff
@@ -1118,6 +1128,41 @@ class ConservativeGC : GC
         stats.freeSize += freeListSize;
         stats.allocatedInCurrentThread = bytesAllocated;
     }
+
+    version (Posix)
+    {
+        // A fork might happen while GC code is running in a different thread.
+        // Because that would leave the GC in an inconsistent state,
+        // make sure no GC code is running by acquiring the lock here,
+        // before a fork.
+
+        extern(C) static void _d_gcx_atfork_prepare()
+        {
+            lockNR();
+        }
+
+        extern(C) static void _d_gcx_atfork_parent()
+        {
+            gcLock.unlock();
+        }
+
+        extern(C) static void _d_gcx_atfork_child()
+        {
+            gcLock.unlock();
```

Comment:
Shouldn't this reinitialize the lock?

Reply:
Sorry. Reinitialize how? I understand that the lock here is a spinlock with yield/sleep backoff, so since it's just a value in memory there's nothing to reinitialize.

Reply:
You are right, SpinLock doesn't have an initialization, but unlock happens to have the same effect. I was rather thinking of a pthread mutex where a handle might be involved.

Reply:
It might be worth looking into switching to a mutex, as I've seen situations where CPU-bound programs are idle in all threads because of the `Thread.sleep(1.msecs)` backoff. But that's something else.